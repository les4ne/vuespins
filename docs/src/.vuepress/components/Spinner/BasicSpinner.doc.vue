<template>
  <div>
    <Spinner />
  </div>
</template>

<script>
import { Spinner } from 'vuespins'

export default {
  name: 'app',
  components: { Spinner }
}
</script>
