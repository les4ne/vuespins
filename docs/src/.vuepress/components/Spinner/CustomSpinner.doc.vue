<template>
  <div>
    <Spinner color="#fb503b" size="80" />
  </div>
</template>

<script>
import { Spinner } from 'vuespins'

export default {
  name: 'app',
  components: { Spinner }
}
</script>
