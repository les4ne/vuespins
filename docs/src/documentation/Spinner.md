# Spinner

## **Basic**

<br>
<client-only>
<Spinner />
</client-only>

### Props

Basic "Spinner" component

| property | propType | required | default | custom |
| :------: | :------: | :------: | :-----: | :----: |
| `color`  |  string  |  false   | #282c34 |   -    |
|  `size`  |  number  |  false   |   50    |   -    |
| `speed`  |  number  |  false   |    2    |   -    |

### Usage

<<< @/src/.vuepress/components/Spinner/BasicSpinner.doc.vue{3,8,10-13}

## **Custom**

<br>
<client-only>
<Spinner color="#fb503b" :size="80" :speed="7" />
</client-only>

### Props

Custom "Spinner" component

| property | propType | required | default | custom  |
| :------: | :------: | :------: | :-----: | :-----: |
| `color`  |  string  |  false   | #282c34 | #fb503b |
|  `size`  |  number  |  false   |   50    |   80    |
| `speed`  |  number  |  false   |    2    |    7    |

### Usage

<<< @/src/.vuepress/components/Spinner/CustomSpinner.doc.vue{3,8,10-13}
