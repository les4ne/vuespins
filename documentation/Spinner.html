<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Spinner | vuespins (1.0.3)</title>
    <meta name="generator" content="VuePress 1.5.4">
    
    <meta name="description" content="Vuespins Docs">
    <meta name="theme-color" content="#3eaf7c">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="preload" href="/vuespins/assets/css/0.styles.954e083e.css" as="style"><link rel="preload" href="/vuespins/assets/js/app.9706235a.js" as="script"><link rel="preload" href="/vuespins/assets/js/2.0c27777a.js" as="script"><link rel="preload" href="/vuespins/assets/js/23.dd6a4554.js" as="script"><link rel="prefetch" href="/vuespins/assets/js/10.eefc7a61.js"><link rel="prefetch" href="/vuespins/assets/js/11.2fa468b3.js"><link rel="prefetch" href="/vuespins/assets/js/12.99a0e334.js"><link rel="prefetch" href="/vuespins/assets/js/13.75ef9ee4.js"><link rel="prefetch" href="/vuespins/assets/js/14.1b89129f.js"><link rel="prefetch" href="/vuespins/assets/js/15.0d87e3f3.js"><link rel="prefetch" href="/vuespins/assets/js/16.5a627578.js"><link rel="prefetch" href="/vuespins/assets/js/17.32f62fe2.js"><link rel="prefetch" href="/vuespins/assets/js/18.fbef641e.js"><link rel="prefetch" href="/vuespins/assets/js/19.7c87094a.js"><link rel="prefetch" href="/vuespins/assets/js/20.369b87e8.js"><link rel="prefetch" href="/vuespins/assets/js/21.47994676.js"><link rel="prefetch" href="/vuespins/assets/js/22.b3c41353.js"><link rel="prefetch" href="/vuespins/assets/js/24.99ee8857.js"><link rel="prefetch" href="/vuespins/assets/js/25.cb1b1be9.js"><link rel="prefetch" href="/vuespins/assets/js/3.48569633.js"><link rel="prefetch" href="/vuespins/assets/js/4.85f34f4c.js"><link rel="prefetch" href="/vuespins/assets/js/5.093d196d.js"><link rel="prefetch" href="/vuespins/assets/js/6.b0142d76.js"><link rel="prefetch" href="/vuespins/assets/js/7.0ed649a5.js"><link rel="prefetch" href="/vuespins/assets/js/8.8f706100.js"><link rel="prefetch" href="/vuespins/assets/js/9.f22a7389.js">
    <link rel="stylesheet" href="/vuespins/assets/css/0.styles.954e083e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/vuespins/" class="home-link router-link-active"><!----> <span class="site-name">vuespins (1.0.3)</span></a> <div class="links"><!----> <nav class="nav-links can-hide"><div class="nav-item"><a href="/vuespins/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://github.com/pwndex/vuespins" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/vuespins/documentation/installation.html" class="nav-link">
  Docs
</a></div> <a href="https://github.com/pwndex/vuespins" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>Documentation</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/vuespins/documentation/Installation.html" class="sidebar-link">Installation</a></li><li><a href="/vuespins/documentation/Ring.html" class="sidebar-link">Ring</a></li><li><a href="/vuespins/documentation/CubeGrid.html" class="sidebar-link">CubeGrid</a></li><li><a href="/vuespins/documentation/Spinner.html" aria-current="page" class="active sidebar-link">Spinner</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vuespins/documentation/Spinner.html#basic" class="sidebar-link">Basic</a></li><li class="sidebar-sub-header"><a href="/vuespins/documentation/Spinner.html#custom" class="sidebar-link">Custom</a></li></ul></li><li><a href="/vuespins/documentation/Chase.html" class="sidebar-link">Chase</a></li><li><a href="/vuespins/documentation/Pulse.html" class="sidebar-link">Pulse</a></li><li><a href="/vuespins/documentation/Square.html" class="sidebar-link">Square</a></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="spinner"><a href="#spinner" class="header-anchor">#</a> Spinner</h1> <h2 id="basic"><a href="#basic" class="header-anchor">#</a> <strong>Basic</strong></h2> <br> <!----> <h3 id="props"><a href="#props" class="header-anchor">#</a> Props</h3> <p>Basic &quot;Spinner&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">-</td></tr> <tr><td style="text-align:center;"><code>speed</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">2</td> <td style="text-align:center;">-</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Spinner</span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> Spinner <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> Spinner <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h2 id="custom"><a href="#custom" class="header-anchor">#</a> <strong>Custom</strong></h2> <br> <!----> <h3 id="props-2"><a href="#props-2" class="header-anchor">#</a> Props</h3> <p>Custom &quot;Spinner&quot; component</p> <table><thead><tr><th style="text-align:center;">property</th> <th style="text-align:center;">propType</th> <th style="text-align:center;">required</th> <th style="text-align:center;">default</th> <th style="text-align:center;">custom</th></tr></thead> <tbody><tr><td style="text-align:center;"><code>color</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">#282c34</td> <td style="text-align:center;">#fb503b</td></tr> <tr><td style="text-align:center;"><code>size</code></td> <td style="text-align:center;">number</td> <td style="text-align:center;">false</td> <td style="text-align:center;">50</td> <td style="text-align:center;">80</td></tr> <tr><td style="text-align:center;"><code>speed</code></td> <td style="text-align:center;">string</td> <td style="text-align:center;">false</td> <td style="text-align:center;">2</td> <td style="text-align:center;">7</td></tr></tbody></table> <h3 id="usage-2"><a href="#usage-2" class="header-anchor">#</a> Usage</h3> <div class="language-vue extra-class"><div class="highlight-lines"><br><br><div class="highlighted"> </div><br><br><br><br><div class="highlighted"> </div><br><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><div class="highlighted"> </div><br><br></div><pre class="language-vue"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>div</span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Spinner</span> <span class="token attr-name">color</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>#fb503b<span class="token punctuation">&quot;</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>80<span class="token punctuation">&quot;</span></span> <span class="token attr-name">speed</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>7<span class="token punctuation">&quot;</span></span> <span class="token punctuation">/&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>div</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>

<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">
<span class="token keyword">import</span> <span class="token punctuation">{</span> Spinner <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">'vuespins'</span>

<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  name<span class="token operator">:</span> <span class="token string">'app'</span><span class="token punctuation">,</span>
  components<span class="token operator">:</span> <span class="token punctuation">{</span> Spinner <span class="token punctuation">}</span>
<span class="token punctuation">}</span>
</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div> <footer class="page-edit"><div class="edit-link"><a href="https://github.com/pwndex/vuespins/edit/master/docs/src/documentation/Spinner.md" target="_blank" rel="noopener noreferrer">Edit this page on GitHub</a> <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></div> <div class="last-updated"><span class="prefix">Last Updated:</span> <span class="time">9/24/2020, 12:53:44 PM</span></div></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/vuespins/documentation/CubeGrid.html" class="prev">
        CubeGrid
      </a></span> <span class="next"><a href="/vuespins/documentation/Chase.html">
        Chase
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/vuespins/assets/js/app.9706235a.js" defer></script><script src="/vuespins/assets/js/2.0c27777a.js" defer></script><script src="/vuespins/assets/js/23.dd6a4554.js" defer></script>
  </body>
</html>
