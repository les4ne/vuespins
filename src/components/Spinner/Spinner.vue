<template>
  <div class="spinner" :style="cssVars">
    <div class="dot1"></div>
    <div class="dot2"></div>
  </div>
</template>

<script>
export default {
  name: '',
  props: {
    color: {
      type: String,
      default: '#333',
    },
    size: {
      type: String,
      default: '50',
    },
  },
  computed: {
    cssVars() {
      return {
        '--vcs-spinner-size': this.size + 'px',
        '--vcs-spinner-color': this.color,
      }
    },
  },
}
</script>

<style scoped>
.spinner {
  width: var(--vcs-spinner-size);
  height: var(--vcs-spinner-size);
  position: relative;
  -webkit-animation: sk-rotate 2s infinite linear;
  animation: sk-rotate 2s infinite linear;
}

.dot1,
.dot2 {
  width: 60%;
  height: 60%;
  display: inline-block;
  position: absolute;
  top: 0;
  background-color: var(--vcs-spinner-color);
  border-radius: 100%;
  -webkit-animation: sk-bounce 2s infinite ease-in-out;
  animation: sk-bounce 2s infinite ease-in-out;
}

.dot2 {
  top: auto;
  bottom: 0;
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

@-webkit-keyframes sk-rotate {
  100% {
    -webkit-transform: rotate(360deg);
  }
}
@keyframes sk-rotate {
  100% {
    transform: rotate(360deg);
    -webkit-transform: rotate(360deg);
  }
}

@-webkit-keyframes sk-bounce {
  0%,
  100% {
    -webkit-transform: scale(0);
  }
  50% {
    -webkit-transform: scale(1);
  }
}

@keyframes sk-bounce {
  0%,
  100% {
    transform: scale(0);
    -webkit-transform: scale(0);
  }
  50% {
    transform: scale(1);
    -webkit-transform: scale(1);
  }
}
</style>