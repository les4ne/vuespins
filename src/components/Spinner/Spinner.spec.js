import { shallowMount } from '@vue/test-utils'
import Spinner from './Spinner.vue'

describe('Spinner.vue', () => {
  it('renders correctly', () => {
    expect(shallowMount(Spinner).element).toMatchSnapshot()
  })
})
