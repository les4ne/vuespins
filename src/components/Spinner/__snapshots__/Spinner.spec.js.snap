// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner.vue renders correctly 1`] = `
<div
  class="spinner"
  style="--vcs-spinner-size: 50px; --vcs-spinner-color: #282c34; --vcs-spinner-speed: 2s; --vcs-spinner-delay: -1s;"
>
  <div
    class="dot1"
  />
   
  <div
    class="dot2"
  />
</div>
`;
